consensus/state_sync_handler.go

```diff
@@ -44,7 +47,7 @@ func (m *consensusModule) handleStateSyncMessage(stateSyncMessage *typesCons.Sta
 		return m.stateSync.HandleStateSyncMetadataRequest(stateSyncMessage.GetMetadataReq())
 	case *typesCons.StateSyncMessage_MetadataRes:
 		m.logger.Info().Str("proto_type", "MetadataResponse").Msg("Handling StateSyncMessage MetadataRes")
-		m.metadataReceived <- stateSyncMessage.GetMetadataRes()
+		m.stateSync.HandleStateSyncMetadataResponse(stateSyncMessage.GetMetadataRes())
```

Comment:
We have two different paradigms here:
1. Sending a message to channel
2. Sending message to function

Can you pick one and be consistent in both?

![Screenshot 2023-05-01 at 8 16 42 PM](https://user-images.githubusercontent.com/1892194/235572363-919f7e42-a26a-4e19-966c-ae14334121ea.png)